packages/auth/src/core/errors.ts

```diff
@@ -0,0 +1,116 @@
+export interface ThirdwebCreateErrorConstructorArgs {
+  code: string;
+  message: string;
+  statusCode: number;
+}
+export interface ThirdwebErrorConstructorArgs<Variables = undefined> {
+  innerError?: Error;
+  variables: Variables;
+}
+
+function formatString(
+  template: string,
+  substitutions: { [key: string]: any }
+): string {
+  return template.replace(/\$\{(\w+)\}/g, (_match, p1) => {
```

Comment:
Would be good to leave a comment on the regex for what it does for people coming back after so it's clear